frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/create-local-volume-set.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from 'react';
+import { match as RouterMatch } from 'react-router';
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  TextInput,
+  Radio,
+  Expandable,
+  TextInputTypes,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import {
+  resourcePathFromModel,
+  BreadCrumbs,
+  Dropdown,
+  resourceObjPath,
+  withHandlePromise,
+  HandlePromiseProps,
+  ButtonBar,
+} from '@console/internal/components/utils';
+import { history } from '@console/internal/components/utils/router';
+import { ListPage } from '@console/internal/components/factory';
+import { k8sCreate, referenceFor } from '@console/internal/module/k8s';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager';
+import { LocalVolumeSetModel } from '../../models';
+import { NodesSelectionList } from './nodes-selection-list';
+import { RowUIDMap, LocalVolumeSetKind } from './types';
+import { getSelectedNodeUIDs } from './utils';
+import './create-local-volume-set.scss';
+
+const VOLUME_TYPE = {
+  SSD: {
+    title: 'SSD / NVMe',
+    property: 'Non Rotational',
+  },
+  HDD: {
+    title: 'HDD',
+    property: 'Rotational',
+  },
+};
+
+const VOLUME_MODE = {
+  Block: {
+    title: 'Block',
+  },
+  Filesystem: {
+    title: 'Filesystem',
+  },
+};
+
+const getDropdownItems = (obj: { [key: string]: { title: string } }) =>
+  Object.keys(obj).reduce((items, type) => {
+    items[type] = obj[type].title;
+    return items;
+  }, {});
+
+const volumeTypeDropdownItems = getDropdownItems(VOLUME_TYPE);
+const volumeModeDropdownItems = getDropdownItems(VOLUME_MODE);
+
+const CreateLocalVolumeSet: React.FC = withHandlePromise<CreateLocalVolumeSetProps>((props) => {
+  const { match, handlePromise, inProgress, errorMessage } = props;
+  const [volumeSetName, setVolumeSetName] = React.useState('');
+  const [storageClassName, setStorageClassName] = React.useState('');
+  const [showNodesList, setShowNodesList] = React.useState(false);
+  const [volumeType, setVolumeType] = React.useState('SSD');
+  const [volumeMode, setVolumeMode] = React.useState('Block');
+  const [maxVolumeLimit, setMaxVolumeLimit] = React.useState('');
+  const [rows, setRows] = React.useState<RowUIDMap>({});
+  const [allSelected, setAllSelected] = React.useState(null);
+
+  const { ns, appName } = match.params;
+  const modelName = LocalVolumeSetModel.label;
+
+  const toggleShowNodesList = () => {
+    setShowNodesList(!showNodesList);
+  };
+
+  const onSubmit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const requestData: LocalVolumeSetKind = {
+      apiVersion: LocalVolumeSetModel.apiVersion,
+      kind: LocalVolumeSetModel.kind,
+      metadata: { name: volumeSetName },
+      spec: {
+        storageClassName,
+        volumeMode,
+        deviceInclusionSpec: {
+          // Only Raw disk supported for 4.5
+          deviceTypes: ['RawDisk'],
```

Comment:
What is this magic string `RawDisk`?

Reply:
This is hard coded intentionally. The support is for `RawDisk` only for only.
Hence, the request data just needs that. No other values are applicable

Reply:
I would prefer it if it came via an enum.